validator/client/beacon-api/activation.go

```diff
@@ -0,0 +1,121 @@
+package beacon_api
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"google.golang.org/grpc"
+)
+
+func (c *beaconApiValidatorClient) waitForActivation(ctx context.Context, in *ethpb.ValidatorActivationRequest) (ethpb.BeaconNodeValidator_WaitForActivationClient, error) {
+	return &waitForActivationClient{
+		ctx:                        ctx,
+		beaconApiValidatorClient:   c,
+		ValidatorActivationRequest: in,
+	}, nil
+}
+
+type waitForActivationClient struct {
+	grpc.ClientStream
+	ctx context.Context
+	*beaconApiValidatorClient
+	*ethpb.ValidatorActivationRequest
+	lastRecvTime time.Time
+}
+
+func computeWaitElements(now, lastRecvTime time.Time) (time.Duration, time.Time) {
+	nextRecvTime := lastRecvTime.Add(time.Duration(params.BeaconConfig().SecondsPerSlot) * time.Second)
+
+	if lastRecvTime.IsZero() {
+		nextRecvTime = now
+	}
+
+	if nextRecvTime.Before(now) {
+		return time.Duration(0), now
+	}
+
+	return nextRecvTime.Sub(now), nextRecvTime
+}
+
+func (c *waitForActivationClient) Recv() (*ethpb.ValidatorActivationResponse, error) {
```

Comment:
if https://github.com/prysmaticlabs/prysm/pull/14514 has issues then this rest implementation needs to be investigated as it does something similar to what that PR implements